src/peer_manager.rs

```diff
@@ -277,9 +279,18 @@ impl PeerManager {
     }
 
     /// Clears the routing table and resets this node's public ID.
-    pub fn reset_routing_table(&mut self, our_public_id: PublicId) {
+    pub fn reset_routing_table(&mut self, our_public_id: PublicId, groups: &[Group]) {
         self.our_public_id = our_public_id;
-        let new_rt = RoutingTable::new(*our_public_id.name(), MIN_GROUP_SIZE);
+        let groups_as_names = groups.into_iter()
+            .map(|&(ref prefix, ref members)| {
+                (*prefix, members.into_iter().map(|pub_id| *pub_id.name()).collect_vec())
+            })
+            .collect_vec();
+        // Nothing can be done to recover from an error here - use unwrap.
```

Comment:
Absolutely! Not sure about that in the first place, since this is an accumulated, trusted message anyway.
We could create a JIRA issue to consider it later.